Tasks/Common/npm-common/npmregistry.ts

```diff
@@ -45,23 +55,29 @@ export class NpmRegistry implements INpmRegistry {
                 username = endpointAuth.parameters['username'];
                 password = endpointAuth.parameters['password'];
                 email = username; // npm needs an email to be set in order to publish, this is ignored on npmjs
+                password64 = (new Buffer(password).toString('base64'));
+
+                auth = nerfed + ":username=" + username + lineEnd;
+                auth += nerfed + ":_password=" + password64 + lineEnd;
+                auth += nerfed + ":email=" + email + lineEnd;
                 break;
-            case 'Token':
+            case 'CrossAccount':
                 email = 'VssEmail';
                 username = 'VssToken';
                 password = endpointAuth.parameters['apitoken'];
+                password64 = (new Buffer(password).toString('base64'));
+
+                auth = nerfed + ":username=" + username + lineEnd;
+                auth += nerfed + ":_password=" + password64 + lineEnd;
+                auth += nerfed + ":email=" + email + lineEnd;
+                break;
+            case 'Token':
+                let apitoken = endpointAuth.parameters['apitoken'];
+                auth = nerfed + ":_authToken=" + apitoken + lineEnd;
                 break;
         }
-       let lineEnd = os.EOL;
-        let nerfed = util.toNerfDart(url); 
-        let password64 = (new Buffer(password).toString('base64'));
-        console.log("##vso[task.setvariable variable=" + endpointId + "BASE64_PASSWORD;issecret=true;]" + password64);
```

Comment:
keeping this console.log is important, it basically says "this string is a password, obfuscate it in any logging"

Reply:
https://github.com/Microsoft/vsts-tasks/blob/master/docs/authoring/commands.md

Reply:
Fixed